UmbracoFood.Tests/Repositories/OrderRepositoryTests.cs

```diff
@@ -0,0 +1,217 @@
+using UmbracoFood.Tests.Repositories.DatabaseFixtures;
+using Xunit;
+using Umbraco.Core.Persistence;
+using System;
+using System.Collections.Generic;
+
+namespace UmbracoFood.Tests.Repositories
+{
+    [Collection("Database collection")]
+    public class OrderRepositoryTests
+    {
+        private readonly OrdersDatabaseFixture _databaseFixture;
+        private readonly Mock<IModelMapper<Order, OrderPoco>> _mapper;
+        private readonly OrderRepository _repo;
+
+        public OrderRepositoryTests(OrdersDatabaseFixture databaseFixture)
+        {
+            _databaseFixture = databaseFixture;
+            var dataBaseProvider = new Mock<IDatabaseProvider>();
+            dataBaseProvider.Setup(dbp => dbp.Db).Returns(_databaseFixture.Db);
+            _mapper = new Mock<IModelMapper<Order, OrderPoco>>();
+
+            _repo = new OrderRepository(dataBaseProvider.Object, _mapper.Object);
+        }
+
+        [Fact]
+        public void EditOrderShouldUpdateImDbOrderAndOrderedMeals()
+        {
+            //arrange
+            var orderId = 1;
+            IList<OrderedMealPoco> orderedMeals = new List<OrderedMealPoco>()
+            {
+                new OrderedMealPoco
+                {
+                    MealName = "meal name",
+                    Price = 16.2,
+                    PurchaserName = "purchaser name"
+                },
+                new OrderedMealPoco
+                {
+                    MealName = "added meal name",
+                    Price = 15.2,
+                    PurchaserName = "purchaser name"
+                }
+            };
+            var orderPocoBeforDbInsert = new OrderPoco
+            {
+                Id = orderId,
+                Deadline = new DateTime(2018, 01, 28, 12, 00, 00),
+                EstimatedDeliveryTime = new DateTime(2016, 03, 28, 13, 00, 00),
+                OrderedMeals = orderedMeals,
+                Owner = "changed owner",
+                RestaurantId = 2,
+                StatusId = (int)OrderStatus.InDelivery
+            };
+
+            _mapper.Setup(m => m.MapToPoco(It.IsAny<Order>())).Returns(orderPocoBeforDbInsert);
+
+            //act
+            _repo.EditOrder(new Order());
+
+            //assert
+            var orderPocoFromDb = GetOrderPocoFromDbById(orderId);
+            Assert.Equal(orderPocoBeforDbInsert.RestaurantId, orderPocoFromDb.RestaurantId);
+            Assert.Equal(orderPocoBeforDbInsert.RestaurantId, orderPocoFromDb.Restaurant.ID);
+            Assert.Equal(orderPocoBeforDbInsert.OrderedMeals.Count, orderPocoFromDb.OrderedMeals.Count);
+            Assert.Equal(orderPocoBeforDbInsert.Owner, orderPocoFromDb.Owner);
+            Assert.Equal(orderPocoBeforDbInsert.StatusId, orderPocoFromDb.StatusId);
+            Assert.Equal(orderPocoBeforDbInsert.StatusId, orderPocoFromDb.Status.Id);
+
+        }
+
+        [Fact]
+        public void RemoveOrderShouldRemoveFromDbOrderAndOrderedMeals()
+        {
+            //arrange
+

```

Comment:
Tutaj nie trzeba najpierw stworzyć Order o ID 5? Bo nawet jeśli potem sprawdzisz czy orderu nie ma, to nie znaczy, że został usunięty